demo/mobile/src/app/modules/sync/syncDataRepository.js

```diff
@@ -0,0 +1,41 @@
+var Promise = require("bluebird");
+
+// Service that is being injected to modules to provide all operations around sync
+function SyncApiDataService(datasetId, syncService) {
+  this.syncManagerPromise = syncService.then(function (managers) {
+    return Promise.promisifyAll(managers[datasetId]);
```

Comment:
This one can't be ported to TS without some refactoring, but that's okay.

Alternative is promisifying methods individually, but more verbose: https://github.com/feedhenry-raincatcher/raincatcher-angularjs/pull/11/files#diff-8e1a8ec6edc1f75011072cb1f15d9cb0R7